floodmodeller_api/toolbox/model_build/structure_log/structure_log.py

```diff
@@ -19,237 +53,445 @@ def _add_fields(self):
             "Weir Coefficient",
             "Culvert Inlet/Outlet Loss",
         ]
-        self._writer.writerow(field)
+        writer.writerow(field)
 
     def _conduit_data(self, conduit):
-        length = 0.0
-        inlet = ""
-        outlet = ""
+        conduit_data = {"length": conduit.dist_to_next}
+        # modified conduit crawler script
+        add_to_conduit_stack = None
+
+        # check if the previous node is an inlet
         previous = self._dat.prev(conduit)
         if hasattr(previous, "subtype") and previous.subtype == "INLET":
-            inlet = previous.ki
+            conduit_data["inlet"] = previous.ki
+
         current_conduit = conduit
-        while current_conduit.dist_to_next != 0:
-            length += current_conduit.dist_to_next
-            current_conduit = self._dat.next(current_conduit)
-        next_conduit = self._dat.next(current_conduit)
-        if hasattr(next_conduit, "subtype") and next_conduit.subtype == "OUTLET":
-            outlet = next_conduit.loss_coefficient
-        return [length, inlet, outlet]
-
-    def _culvert_loss_data(self, inlet, outlet):
-        culvert_loss = ""
-        if inlet != "" and outlet != "":
-            culvert_loss = f"Ki: {inlet}, Ko: {outlet}"
-        elif inlet != "":
-            culvert_loss = f"Ki: {inlet}"
-        elif outlet != "":
-            culvert_loss = f"Ko: {outlet}"
-        return culvert_loss
-
-    def _circular_data(self, conduit, length):
-        dimensions = f"dia: {conduit.diameter:.2f} x l: {length:.2f}"
-        all_mannings = [
+        # check if the
```

Comment:
?

Reply:
addressed, and added/clarified other comments in the area